### PR TITLE
Document using round-robin DNS for discovery

### LINKED ISSUE #8427: DNS round robin assisted discovery
If all nodes in a cluster are added to a round robin DNS record and the following changes are made:
- `discovery.zen.ping.unicast.hosts` has the DNS RR name and Zen tries connecting to all the IPs 
- clients initially connect to the cluster via the DNS RR name

Why? Modern HTTP clients will get a random node in the cluster, but if it fails they'll automatically try the rest until one works [1], when it can then discover the rest of the cluster in the normal way. Currently AFAIK to guarantee a client will always work on startup (ie. before it has a chance to discover all the other nodes in the cluster) and for unicast zen to always work, it has to be supplied with a _current_ list of at least half (?) of the master-eligible nodes.

This would make the cluster rely on DNS to operate, although I expect most people refer to their nodes via DNS anyway, so doesn't add any further dependency for people who'd want to take advantage of this.

[1] http://blog.engelke.com/2011/06/07/web-resilience-with-round-robin-dns/

### PR DESCRIPTION
**Pull request by clintongormley**
Closes #8427

### CODE REVIEW

File: docs/reference/modules/network.asciidoc [hunk 1]
```diff
@@ -31,7 +31,9 @@ Defaults to `_local_`.
 In order to join a cluster, a node needs to know the hostname or IP address of
 at least some of the other nodes in the cluster.  This setting provides the
 initial list of other nodes that this node will try to contact. Accepts IP
-addresses or hostnames.
+addresses or hostnames.  If a hostname lookup resolves to multiple IP addresses --
```
Comment (bleskes) [inline docs/reference/modules/network.asciidoc]: while `If a hostname lookup resolves to multiple IP addresses ... each IP will be used..` is true, how does it relate to round robin DNS ? it seems that one just returns one, albeit a different one on each resolving round?

**Comment by clintongormley** [comment]
@jasontedor after the change to resolve unicast hosts lazily, does the discovery process still work as documented in this PR?

File: docs/reference/modules/network.asciidoc [hunk 2]
```diff
@@ -31,7 +31,11 @@ Defaults to `_local_`.
 In order to join a cluster, a node needs to know the hostname or IP address of
 at least some of the other nodes in the cluster.  This setting provides the
 initial list of other nodes that this node will try to contact. Accepts IP
-addresses or hostnames.
+addresses or hostnames.  If a hostname lookup resolves to multiple IP
+addresses then each IP address will be used for discovery.
+https://en.wikipedia.org/wiki/Round-robin_DNS[Round robin DNS] -- returning a
+different IP from a list on each lookup -- can be used for discovery; non-
+existent IP addresses will throw exceptions and cause another DNS lookup.
```
Comment (jasontedor) [inline docs/reference/modules/network.asciidoc]: `lookup.` -> `lookup on the next round of pinging (subject to JVM DNS caching).`

**Review by jasontedor — APPROVED** [review]
I left one comment.

File: docs/reference/modules/network.asciidoc (hunk 2 above)
Comment (clintongormley) [inline docs/reference/modules/network.asciidoc]: thanks @jasontedor - included your suggestion

**Pull request merged**